### PR TITLE
Fixed #19878 -- Deprecated TemplateView passing URL kwargs into context.

### PR DESCRIPTION
**Pull request by adamchainz**
[Ticket #19878](https://code.djangoproject.com/ticket/19878)

### CODE REVIEW

**Comment by adamchainz** [comment]
This is still a bit of WIP - I need to follow the deprecation guidelines and come up with a better message.

**Comment by step21** [comment]
Thanks. Let me know if you need help with something or documentation or whatever.

File: django/views/generic/base.py [hunk 1]
```diff
@@ -152,10 +155,24 @@ def get_template_names(self):
...
 class TemplateView(TemplateResponseMixin, ContextMixin, View):
     """
-    Render a template. Pass keyword arguments from the URLconf to the context.
+    Render a template.
+
+    Deprecated: Pass keyword arguments from the URLconf to the context.
     """
     def get(self, request, *args, **kwargs):
-        context = self.get_context_data(**kwargs)
+        context_kwargs = {}
```
Comment (carltongibson) [inline django/views/generic/base.py]: Can you put this block in a helper?

```
context_kwargs = _adams_helper_with_a_good_name(...)
```

Comment (adamchainz) [reply django/views/generic/base.py]: 👌 Decided not to put my name in the function signature.

**Comment by felixxm** [comment]
@adamchainz Is this ready for a review? (release notes are missing).

**Comment by adamchainz** [comment]
Thanks for the bump @felixxm 

Came back to it and fixed a pretty vicious bug with the closure binding, hence moved the test to use two variables.

Added a release note.

**Comment by felixxm** [comment]
@adamchainz Thanks :+1: I moved deprecation tests to a separate class, simplified them and pushed minor edits to the docs.

**Comment by adamchainz** [comment]
All looks good 👍

**Pull request merged**